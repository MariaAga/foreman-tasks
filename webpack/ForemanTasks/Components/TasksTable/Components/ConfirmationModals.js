import React from 'react';
import PropTypes from 'prop-types';
import {
  CANCEL_CONFIRM_MODAL_ID,
  RESUME_CONFIRM_MODAL_ID,
  CANCEL_SELECTED_CONFIRM_MODAL_ID,
  RESUME_SELECTED_CONFIRM_MODAL_ID,
} from '../TasksTableConstants';
import { CancelConfirm } from './CancelConfirm';
import { ResumeConfirm } from './ResumeConfirm';

export const ConfirmationModals = ({
  modalProps,
  tasksActions,
  selectedRowsLen,
}) => (
  <React.Fragment>
    <CancelConfirm
      closeModal={modalProps.cancelModal.setModalClosed}
      action={tasksActions.cancelTask}
      selectedRowsLen={1}
      id={CANCEL_CONFIRM_MODAL_ID}
    />
    <ResumeConfirm
      closeModal={modalProps.resumeModal.setModalClosed}
      action={tasksActions.resumeTask}
      selectedRowsLen={1}
      id={RESUME_CONFIRM_MODAL_ID}
    />
    <CancelConfirm
      closeModal={modalProps.cancelSelectedModal.setModalClosed}
      action={tasksActions.cancelSelectedTasks}
      selectedRowsLen={selectedRowsLen}
      id={CANCEL_SELECTED_CONFIRM_MODAL_ID}
    />
    <ResumeConfirm
      closeModal={modalProps.resumeSelectedModal.setModalClosed}
      action={tasksActions.resumeSelectedTasks}
      selectedRowsLen={selectedRowsLen}
      id={RESUME_SELECTED_CONFIRM_MODAL_ID}
    />
  </React.Fragment>
);

ConfirmationModals.propTypes = {
  modalProps: PropTypes.object.isRequired,
  selectedRowsLen: PropTypes.number.isRequired,
  tasksActions: PropTypes.shape({
    cancelTask: PropTypes.func,
    resumeTask: PropTypes.func,
    cancelSelectedTasks: PropTypes.func,
    resumeSelectedTasks: PropTypes.func,
  }).isRequired,
};

export default ConfirmationModals;
