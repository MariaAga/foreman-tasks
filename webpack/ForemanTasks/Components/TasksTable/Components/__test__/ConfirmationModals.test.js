import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { ConfirmationModals } from '../ConfirmationModals';

const fixtures = {
  'renders ConfirmationModals': {
    modalProps: {
      cancelSelectedModal: { setModalClosed: jest.fn() },
      resumeSelectedModal: { setModalClosed: jest.fn() },
      cancelModal: { setModalClosed: jest.fn() },
      resumeModal: { setModalClosed: jest.fn() },
    },
    selectedRowsLen: 7,
    tasksActions: {
      cancelTask: jest.fn(),
      resumeTask: jest.fn(),
      cancelSelectedTasks: jest.fn(),
      resumeSelectedTasks: jest.fn(),
    },
  },
};

describe('ConfirmationModals', () =>
  testComponentSnapshotsWithFixtures(ConfirmationModals, fixtures));
