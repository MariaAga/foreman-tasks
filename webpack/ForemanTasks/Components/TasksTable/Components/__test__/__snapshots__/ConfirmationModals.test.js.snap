// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModals renders ConfirmationModals 1`] = `
<Fragment>
  <CancelConfirm
    action={[MockFunction]}
    closeModal={[MockFunction]}
    id="cancelConfirmModal"
    selectedRowsLen={1}
  />
  <ResumeConfirm
    action={[MockFunction]}
    closeModal={[MockFunction]}
    id="resumeConfirmModal"
    selectedRowsLen={1}
  />
  <CancelConfirm
    action={[MockFunction]}
    closeModal={[MockFunction]}
    id="cancelSelectedConfirmModal"
    selectedRowsLen={7}
  />
  <ResumeConfirm
    action={[MockFunction]}
    closeModal={[MockFunction]}
    id="resumeSelectedConfirmModal"
    selectedRowsLen={7}
  />
</Fragment>
`;
